### PR TITLE
Paddle Op conversion for SSD-MobileNetV3: exp, stack

### PR DESCRIPTION
**Pull request by liubo-intel**
### Details:
 - *PaddlePaddle op convert for "[exp](https://www.paddlepaddle.org.cn/documentation/docs/en/1.7/api/layers/exp.html)" and "[stack](https://www.paddlepaddle.org.cn/documentation/docs/en/1.7/api/layers/stack.html#stack)", which is required by SSD-MobileNetV3
 

### Tickets:
 - *63557*

### CODE REVIEW

**Comment by liubo-intel** [comment]
Hi, @ceciliapeng2011 @luo-cheng2021 @ilyachur @mvafin : 
could you please help review this PaddlePaddle Op conversion PR when you are free? tks~

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+NamedOutputs stack(const NodeContext& node) {
+    auto datas = node.get_ng_inputs("X");
+    int axis = 0;
+    if (node.has_attribute<int32_t>("axis"))
+        axis = node.get_attribute<int32_t>("axis");
```
Comment (luo-cheng2021) [inline ngraph/frontend/paddlepaddle/src/op/stack.cpp:16]: 14~16 could use 'auto axis = node.get_attribute<int32_t>("axis", 0);'

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    for (const auto& data : datas) {
+        PDPD_OP_VALIDATION_CHECK(node, data.get_partial_shape().rank().is_static(), "stack rank must be static");
+        PDPD_OP_VALIDATION_CHECK(node,
+                                 data_shape == data.get_partial_shape(),
+                                 "stack input tensor must the same shape!");
+        PDPD_OP_VALIDATION_CHECK(node,
+                                 data_type == data.get_element_type(),
+                                 "stack input tensor must the same data types!");
+
+        node_datas_reshape.push_back(std::make_shared<opset8::Unsqueeze>(data, axis_const));
```
Comment (luo-cheng2021) [inline ngraph/frontend/paddlepaddle/src/op/stack.cpp:35]: Unsqueeze supports negative axis and may also remove the check between line 27~33 to support dynamic shape, Concat should do the check itself.

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp (hunk 1 above)
Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp]: Done

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp (hunk 2 above)
Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:29]: removed the static shape check, while kept the original shape and element type check of this op.

File: ngraph/frontend/paddlepaddle/src/op/exp.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#include <ngraph/opsets/opset8.hpp>
+#include <node_context.hpp>
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+NamedOutputs exp(const NodeContext& node) {
+    auto data = node.get_ng_input("X");
+    PDPD_OP_VALIDATION_CHECK(node, data.get_element_type() != ov::element::f64, "exp not support fp64 input type !");
```
Comment (mvafin) [inline ngraph/frontend/paddlepaddle/src/op/exp.cpp:14]: This should not be done in the frontend. If fp64 is not supported in inference_engine that is not a problem of a frontend.  If you observe test fails you can create ticket for this problem to be fixed on IE side.

**Review by mvafin — CHANGES_REQUESTED** [review]

File: ngraph/frontend/paddlepaddle/src/op/exp.cpp (hunk 3 above)
Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/exp.cpp]: have removed the check. it's a common limitation of IE, so maybe we add this limitation description to  'paddle op conversion document' later?

Comment (ceciliapeng2011) [reply ngraph/frontend/paddlepaddle/src/op/exp.cpp]: agree

File: ngraph/frontend/paddlepaddle/src/op/exp.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <ngraph/opsets/opset8.hpp>
```
Comment (ceciliapeng2011) [inline ngraph/frontend/paddlepaddle/src/op/exp.cpp:5]: better follow https://github.com/openvinotoolkit/openvino/blob/master/ngraph/frontend/paddlepaddle/src/op/gelu.cpp#L7

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+NamedOutputs stack(const NodeContext& node) {
+    auto datas = node.get_ng_inputs("X");
+    auto axis = node.get_attribute<int32_t>("axis", 0);
+    auto data_shape = datas[0].get_partial_shape();
+    auto data_type = datas[0].get_element_type();
+    OutputVector node_datas_reshape;
+
+    auto axis_const = std::make_shared<opset8::Constant>(element::i64, Shape{}, axis);
+    for (const auto& data : datas) {
+        if (data.get_partial_shape().rank().is_static())
```
Comment (ceciliapeng2011) [inline ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: @luo-cheng2021 do you also think we need this check here? Can not concat support broadcast?

Comment (luo-cheng2021) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: I think can be removed here

Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: Hi, @ceciliapeng2011 and @luo-cheng2021 :  in paddle [spec of 'stack' op](https://www.paddlepaddle.org.cn/documentation/docs/en/1.7/api/layers/stack.html#stack), we can find that " If x is a list, **the shapes of all these Tensors must be the same.** ", so do you think we'd better keep this check to align with its spec?

Comment (ceciliapeng2011) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: @liubo-intel  Paddle's check have already guaranteed that this negative case won't happen. That means, no such paddle model can be generated. You can have a try. 

On the other hand, does openvino spec of 'concat' have this limitation as well? If not, we can ignore this check. Even if it is, can we leave it to other modules to throw this kind of exception? We cannot guarantee the case of dynamic shape, anyway.

Comment (ceciliapeng2011) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: @mvafin  @ilyachur @zhangYiIntel  look forward to your opinion on this check. thanks.

Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: > @liubo-intel Paddle's check have already guaranteed that this negative case won't happen. That means, no such paddle model can be generated. You can have a try.
> 
> On the other hand, does openvino spec of 'concat' have this limitation as well? If not, we can ignore this check. Even if it is, can we leave it to other modules to throw this kind of exception? We cannot guarantee the case of dynamic shape, anyway.

agree that, if we can guarantee that no such negative paddle model case,  this check can be removed.

File: ngraph/frontend/paddlepaddle/src/op/exp.cpp (hunk 4 above)
Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/exp.cpp]: modified

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+namespace op {
+using namespace default_opset;
+NamedOutputs stack(const NodeContext& node) {
+    auto datas = node.get_ng_inputs("X");
+    auto axis = node.get_attribute<int32_t>("axis", 0);
+    auto data_shape = datas[0].get_partial_shape();
+    auto data_type = datas[0].get_element_type();
+    OutputVector node_datas_reshape;
+
+    auto axis_const = std::make_shared<Constant>(element::i64, Shape{}, axis);
```
Comment (zhangYiIntel) [inline ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: add range check for `axis` to ensure it locates inside  `[−(R+1),R+1]`, where `R` is the dimension of the first input

**Review by zhangYiIntel — CHANGES_REQUESTED** [review]

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp (hunk 6 above)
Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:21]: done

**Review by zhangYiIntel — APPROVED** [review]

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            "axis range is [-(R+1), R+1)!");
+
+    for (const auto& data : datas) {
+        if (data.get_partial_shape().rank().is_static())
+            PDPD_OP_VALIDATION_CHECK(node,
+                                     data_shape == data.get_partial_shape(),
+                                     "stack input tensor must the same shape!");
+        PDPD_OP_VALIDATION_CHECK(node,
+                                 data_type == data.get_element_type(),
+                                 "stack input tensor must the same data types!");
```
Comment (mvafin) [inline ngraph/frontend/paddlepaddle/src/op/stack.cpp:35]:
```suggestion
                                 "stack input tensor must have the same data types!");
```

File: ngraph/frontend/paddlepaddle/src/op/stack.cpp (hunk 5 above)
Comment (mvafin) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp:29]: I agree that this check is not needed. All required checks will be done when ngraph op is created.

Comment (liubo-intel) [reply ngraph/frontend/paddlepaddle/src/op/stack.cpp]: have removed this check.

**Comment by liubo-intel** [comment]
Hi, @mvafin and @ilyachur :  if nothing further modification needed for this PR, could you please help merge it? tks :)

**Review by mvafin — APPROVED** [review]

**Pull request merged**